lib/Page.js

```diff
@@ -585,7 +587,7 @@ class Page extends EventEmitter {
 
     const buffer = new Buffer(result.data, 'base64');
     if (options.path)
-      fs.writeFileSync(options.path, buffer);
+      await writeFileAsync(options.path, buffer);
```

Comment:
Why is there even an option to write the file here? Either puppeteer should return the actual stream or users should just write the buffer themselves.

Reply:
@graingert two reasons:
- it comes handy from user standpoint: the majority of `page.screenshot` usages have `path` option in it.
- it's cheap from the project standpoint.